plugins/modules/packaging/os/pacman_key.py

```diff
@@ -0,0 +1,308 @@
+        tmpfile = os.fdopen(tmpfd, "w")
+        tmpfile.write(data)
+        tmpfile.close()
+        return tmpname
+
+    def add_key(self, keyring, keyfile, keyid, fingerprint=None):
+        """Add key to pacman's keyring"""
+        cmd = [self.pacman_key, '--gpgdir', keyring]
+        if fingerprint is not None:
+            self.verify_keyfile(keyfile, keyid, fingerprint)
+        self.module.run_command(cmd + ['--add', keyfile], check_rc=True)
+        self.lsign_key(keyid, keyring)
+
+    def remove_key(self, keyring, keyid):
+        """Remove key from pacman's keyring"""
+        cmd = [self.pacman_key, '--gpgdir', keyring]
+        self.module.run_command(cmd + ['--delete', keyid], check_rc=True)
+
+    def verify_keyfile(self, keyfile, keyid, fingerprint):
+        """Verify that keyfile matches the specified keyid & fingerprint"""
+        if keyfile is None:
+            self.module.fail_json(msg="expected a key, got none")
+        elif keyid is None:
+            self.module.fail_json(msg="expected a keyid, got none")
+        elif fingerprint is None:
+            self.module.fail_json(msg="expected a fingerprint, got none")
+
+        _, stdout, _ = self.module.run_command([self.gpg, '--with-colons', '--with-fingerprint', '--batch', '--no-tty', '--show-keys', keyfile], check_rc=True)
+
+        extracted_keyid = extracted_fingerprint = None
+        for line in stdout.splitlines():
+            if line.startswith('pub:'):
+                extracted_keyid = line.split(':')[4]
+            elif line.startswith('fpr:'):
+                extracted_fingerprint = line.split(':')[9]
+                break
+
+        if extracted_keyid != keyid:
+            self.module.fail_json(msg="keyid does not match. expected %s, got %s" % (keyid, extracted_keyid))
+        elif extracted_fingerprint != fingerprint:
+            self.module.fail_json(msg="fingerprint does not match. expected %s, got %s" % (fingerprint, extracted_fingerprint))
+
+    def key_in_keyring(self, keyring, keyid):
+        "Check if the keyid is in pacman's keyring"
+        rc, _, stderr = self.module.run_command(
+            [self.gpg, '--with-colons', '--batch', '--no-tty',
+                '--no-default-keyring', '--keyring=' + keyring + '/pubring.gpg',
+                '--list-keys', keyid])
+        if rc != 0:
+            if stderr.find("No public key") >= 0:
+                return False
+            else:
+                self.module.fail_json(msg="gpg returned an error: %s" % stderr)
+        return True
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            id=dict(type='str', required=True),
+            data=dict(type='str'),
+            file=dict(type='path'),
+            url=dict(type='str'),
+            keyserver=dict(type='str'),
+            fingerprint=dict(type='str'),
+            force_update=dict(type='bool', default=False),
+            keyring=dict(type='path', default='/etc/pacman.d/gnupg'),
```

Comment:
```suggestion
            keyring=dict(type='path'),
```
Alternatively, add the default in the documentation as well.